Gems/RecastNavigation/Code/Source/Misc/RecastNavigationMeshComponentController.cpp

```diff
@@ -63,6 +63,12 @@ namespace RecastNavigation
 
     void RecastNavigationMeshComponentController::UpdateNavigationMeshBlockUntilCompleted()
     {
+        bool notInProgress = false;
```

Comment:
> question: Is it possible to rewrite this as inProgress = true as opposed to the double negative of this not not being in progress 😅

That wouldn't work because the `expected` value in `compare_exchange_strong(expected, desired)` is false, it cannot be switched to true.

But an improvement could be to declare the variable as const and inside the `if` to reduce its scope:

````
// If not in progress then switch it to in progress.
// If it didn't switch it to in progress, return.
if (const bool notInProgress = false;
    !m_updateInProgress.compare_exchange_strong(notInProgress, true))
{
    return;
}
````

similar can be done in others.

Reply:
Argh that's a pain, fair enough, just takes a little more mental gymnastics to parse